teuthology/__init__.py

```diff
@@ -37,10 +37,10 @@
 # do our best, but if it fails, continue with above
 
 try:
-    __version__ += '-' + subprocess.check_output(
+    __version__ += '-' + str(subprocess.check_output(
```

Comment:
why do we need `str` here?

Reply:
this is early version of *six.ensure_str* ;-)

Reply:
i think it's just more difficult to identify the compatibility changes from the functional code, if we choose to keep the homebrew solution.